src/main/java/org/wahlzeit_revisited/service/UserService.java

```diff
@@ -0,0 +1,74 @@
+package org.wahlzeit_revisited.service;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import jakarta.ws.rs.core.Response;
+import org.wahlzeit.services.SysLog;
+import org.wahlzeit_revisited.auth.AccessRights;
+import org.wahlzeit_revisited.dto.UserDto;
+import org.wahlzeit_revisited.model.User;
+import org.wahlzeit_revisited.model.UserFactory;
+import org.wahlzeit_revisited.repository.UserRepository;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Singleton
+public class UserService {
+
+    @Inject
+    public Transformer transformer;
+    @Inject
+    public UserRepository repository;
+    @Inject
+    public UserFactory factory;
+
+    /*
+     * business methods
+     */
+
+    public Response getUsers() throws SQLException {
+        List<User> userList = repository.findAll();
+
+        List<UserDto> responseDto = new ArrayList<>(userList.size());
+        for (User photo : userList) {
+            responseDto.add(transformer.transform(photo));
+        }
+        return Response.ok(responseDto).build();
+    }
+
+    public synchronized Response createUser(String username, String email, String plainPassword) throws SQLException {
```

Comment:
A `synchronized` was added and the Service is now annotated as a `Singleton` as with this pattern, there is a race condition between `repository::hasByEmail(String email)` and `repositoy::insert(User user)`, as the insert operation is no longer ACID.